ProjectSourceCode/views/partials/profile.hbs

```
@@ -5,26 +5,25 @@
     <div class="mb-6">
         <label class="block text-gray-700 font-semibold mb-2">Profile Picture</label>
         <div class="flex items-center space-x-4">
+            <!-- Hidden File Input -->
             <input 
                 type="file" 
                 name="profilePicture" 
                 id="profilePicture" 
                 accept="image/*" 
                 class="hidden"
             >
+            <!-- Profile Picture Label (Clickable) -->
             <label 
                 for="profilePicture" 
                 class="w-24 h-24 rounded-full bg-gray-200 flex items-center justify-center cursor-pointer 
-                        hover:opacity-75 transition-opacity relative group"
+                    hover:opacity-75 transition-opacity relative group overflow-hidden"
             >
-                <svg class="w-12 h-12 text-gray-400 z-10" fill="currentColor" viewBox="0 0 24 24">
+                <!-- Default Icon or Image Preview -->
+                <img id="profilePreview" {{#if user.profile_photo_path}}src={{user.profile_photo_path}}{{/if}} class="absolute inset-0 w-full h-full object-cover rounded-full {{#if user.profile_photo_path}}{{else}}hidden{{/if}}">
+                <svg id="defaultIcon" class="w-12 h-12 text-gray-400 z-10 {{#if user.profile_photo_path}}hidden{{/if}}" fill="currentColor" viewBox="0 0 24 24">
```

Do the conditionals not do the same thing as line 58 and 59 in app.js?

---

Hmm, I don't think so. Lines 58 and 59 only matter at all when you are submitting your profile pic for the first time (there is no existing one in the database). The conditionals take care of every other case, but in that one edge case lines 58 and 59 swap which one is hidden because otherwise you can't preview your first profile picture.

---

Because when you have no existing photo in the database, the conditionals will show the default and hide the preview, then, when you upload one but don't submit yet, the function in app.js changes the source of the preview, but it still wont show it to the user because the conditionals hid it originally. That's where lines 58 and 59 come in to show it and hide the default.

---

Oooohhhhh okay! Cool!